### ISSUE TITLE
ExecutorWebDriver

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#12374]
We should stop relying on Selenium for Chrome/Edge/Safari support, and instead use WebDriver directly.

Relying on Selenium adds further levels of indirection (and the potential to abstract away bugs) and makes it harder to use custom WebDriver extensions for testing.

- [x] Support WebDriver as an option (#12380)
- [x] Enable WebDriver by default for Chrome (#13200, #13225)
- [ ] Enable WebDriver by default for Edge
- [x] Enable WebDriver by default for Safari (#14213)
- [ ] Enable WebDriver by default for Chrome on Android
- [ ] Enable WebDriver by default for Opera
- [x] Enable WebDriver by default for WebKit (#13339)

**Comment by foolip** [comment]
Previous PR at https://github.com/web-platform-tests/wpt/pull/10197.

**Comment by foolip** [comment]
New PR at https://github.com/web-platform-tests/wpt/pull/12380.

**Comment by foolip** [comment]
Reopening to also track this being the default. @gsnedders what's the plan for that?

**Comment by gsnedders** [comment]
I've been trying to get Windows CI up and running again, and got somewhat side-tracked. :)

I think the plan is to just try landing it, at this point?

**Comment by gsnedders** [comment]
#13200

**Comment by gsnedders** [comment]
So, one the above to-do is done, this leaves just IE and Sauce; I think we should probably just drop support for everything except wdspec tests for IE (cc @jimevans), given almost all will just break horribly, and Sauce we can't do anything about until they support the W3C protocol.

**Comment by foolip** [comment]
@gsnedders, can I suggest comparing Taskcluster runs before/after the change, finding the commit where this landed in https://staging.wpt.fyi/test-runs?label=taskcluster. Example diff URL:
https://staging.wpt.fyi/results/?product=firefox@59bed3508c&product=firefox@ddb3e30ef4&diff=true

**Comment by foolip** [comment]
Also, a similar comparison for the Buildbot runs on prod would be good. Although that'll also include test changes as it's not run per push to master.

**Comment by foolip** [comment]
https://staging.wpt.fyi/results/?product=chrome@9bc71d422a&product=chrome@1e7408fcd3&diff=true

**Comment by foolip** [comment]
@gsnedders, Safari 12.0 was just released with macOS 10.14 Mojave, can you check if the accompanying SafariDriver is updated so that we can use ExecutorWebDriver?

**Comment by foolip** [comment]
Excellent! @gsnedders, can you give it a try to enable, and coordinate with @jugglinmike who is trying to set up runs for Safari stable?

**Comment by burg** [comment]
> @foolip In my experience, the driver shipped with Safari 12 uses the W3C protocol dialect by default. @burg would be able to say for sure, but modulo bugs in the implementation, it should be usable with any spec-compliant local end implementation.

This is correct. JSONWP has been dropped entirely in safaridriver for Safari Technology Preview.

**Comment by gsnedders** [comment]
@jugglinmike do you want to try flipping us over to safari_webdriver for a build or two on wpt.fyi? or should we try (pretty blindly!) to do that upstream?

**Comment by foolip** [comment]
I've tried enabling safari_webdriver in https://github.com/foolip/wpt/pull/5:
https://travis-ci.org/foolip/wpt/jobs/439114299

So at least the infrastructure/ tests work. Also curious to hear how @jugglinmike would like to roll this out. I suspect it depends on the status of Safari 12 runs?

**Comment by jugglinmike** [comment]
Thanks for the offer, @gsnedders. I would like to take you up on that, but I'm still working through [an intermittent error in testing Safari](https://github.com/web-platform-tests/results-collection/issues/615).  Resolving that is my top priority, and I hope to have it licked before the end of the week. You can refer to that issue for progress updates.

**Comment by mdittmer** [comment]
_Ping from your friendly neighbourhood ecosystem infra rotation_

Any updates on this issue, @gsnedders?

**Comment by gsnedders** [comment]
* Edge is blocked on wpt.fyi moving to Windows 10 RS5 (and potentially on it being rolled out more generally to users)
 * Opera is blocked on OperaDriver working in W3C mode (webdriver/tests/new_session/ totally fails)
 * Chrome on Android is blocked on someone actually having something set up to test it

**Comment by foolip** [comment]
@gsnedders ExecutorWebDriver has been a thing for a while now. Can you mark the important follow-up as roadmap priority and close this issue?
